### PR TITLE
add download option from dump functionality

### PR DESCRIPTION
**Pull request by tino097**
Enables users to choose to what format resource will be downloaded

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

File: ckan/templates/package/resource_read.html [hunk 1]
```diff
@@ -43,6 +44,24 @@
                     <i class="fa fa-arrow-circle-o-down"></i> {{ _('Download') }}
                   {% endif %}
                 </a>
+                {%if res.url_type == 'upload' and res.datastore_active %}
```
Comment (amercader) [inline ckan/templates/package/resource_read.html]: Don't check for `res.url_type == 'upload'` you can have stuff in the DataStore that hasn't been uploaded

**Comment by wardi** [comment]
No need for the upload check and make sure this is inside a block so it can be overridden

**Comment by TkTech** [comment]
Better yet , can we get away with no hardcoding at all? There should be an action method for DataStore to advertise supported formats.

File: ckan/templates/package/resource_read.html [hunk 2]
```diff
@@ -43,6 +44,25 @@
                     <i class="fa fa-arrow-circle-o-down"></i> {{ _('Download') }}
                   {% endif %}
                 </a>
+                 {% block donwload_resource_button %}
```
Comment (wardi) [inline ckan/templates/package/resource_read.html]: download_resource_button

File: ckan/templates/package/resource_read.html [hunk 3]
```diff
@@ -43,6 +44,25 @@
...
+                  {%if res.datastore_active %}
+                <button class="btn btn-primary dropdown-toggle" data-toggle="dropdown">
+                    <span class="caret"></span>
+                  </button>
+                <ul class="dropdown-menu">
+                  <li>
+                    <a href="{{ h.url_for(controller='ckanext.datastore.controller:DatastoreController', action='dump', resource_id=res.id) }}"
+                      target="_blank"><span>CSV</span></a>
```
Comment (wardi) [inline ckan/templates/package/resource_read.html]: Maybe default to bom=True for CSV and TSV so that Excel can load them properly

**Comment by wardi** [comment]
@TkTech Having a block that can be overridden is our best option for now. That lets sites define whichever options make sense for them. Also the list of formats datastore supports exporting isn't extensible with a plugin interface yet.

**Pull request merged**